### PR TITLE
 Solved Padding issue in header

### LINKED ISSUE #8134: Padding Bug in CMS Editor
Screenshot 2022-03-16 at 13 34 08

Search logo is not aligned, also right button and left text is not vertically aligned.

### PR DESCRIPTION
**Pull request by Marmik2003**
Solved search icon padding and plus icon padding in header.
Made them vertically aligned
Fixes #8134

### CODE REVIEW

File: client/scss/components/_button.scss [hunk 1]
```diff
@@ -164,7 +164,10 @@
 
       .icon {
         @include svg-icon(1rem);
-        padding: 0.75em;
+        padding: 0.36em;
+        @media screen and (min-width: $screen-sm-min) {
+          padding: 0.75em;
+        }
```
Comment (lb-) [inline client/scss/components/_button.scss:170]: is there a way we can solve this without adding a new screen width?

not sure as I have not run the code myself, but maybe the line-height approach from https://css-tricks.com/centering-css-complete-guide/

or flexbox?

Comment (lb-) [inline client/scss/components/_button.scss:170]: is there a way we can solve this without adding a new screen width?

not sure as I have not run the code myself, but maybe the line-height approach from https://css-tricks.com/centering-css-complete-guide/

or flexbox?

**Review by PaarthAgarwal — COMMENTED** [review]
@lb-, please check if the issue of the misaligned plus icon (of add images) exists in the first place. I tried the steps to replicate the behavior as mentioned in #8134 but the icon appeared well aligned to me.
If that's the case, we can modify the search icon alone, leaving the plus icon untouched.

**Comment by Marmik2003** [comment]
> @lb-, please check if the issue of the misaligned plus icon (of add images) exists in the first place. I tried the steps to replicate the behavior as mentioned in #8134 but the icon appeared well aligned to me. If that's the case, we can modify the search icon alone, leaving the plus icon untouched.

@PaarthAgarwal The logo is showing completely fine, but below the width of 800px, it becomes misaligned...

**Comment by PaarthAgarwal** [comment]
> > @lb-, please check if the issue of the misaligned plus icon (of add images) exists in the first place. I tried the steps to replicate the behavior as mentioned in #8134 but the icon appeared well aligned to me. If that's the case, we can modify the search icon alone, leaving the plus icon untouched.
> 
> @PaarthAgarwal The logo is showing completely fine, but below the width of 800px, it becomes misaligned...

Thanks for clarifying @Marmik2003!

File: client/scss/components/_button.scss [hunk 2]
```diff
@@ -164,7 +164,9 @@
 
       .icon {
         @include svg-icon(1rem);
-        padding: 0.75em;
+        position: absolute;
```
Comment (stevedya) [inline client/scss/components/_button.scss:167]: Nice work @Marmik2003! 

**Suggestion:** We could also get away without positioning absolute on this one.
The issue seems to only be happening in elements wrapped with `.icon-wrapper` above. If we add some flex-box centering to that instead it should also resolve the issue. 

e.g 
```
.icon-wrapper {
 display: flex;
 justify-content: center;
 align-items: center;
}
```

File: client/scss/components/_header.scss [hunk 3]
```diff
@@ -56,6 +56,9 @@ header {
     // stylelint-disable-next-line property-disallowed-list
     left: 0.5em;
     inset-inline-start: 0.5em;
+
+    // Icon Positioning (Vertically Centered)
+    padding: 10px 0;
```
Comment (stevedya) [inline client/scss/components/_header.scss:61]: **Suggestion:** We could also align vertically using 50% of the height of the element to ensure they will always be centered, even if the input was to change in size a bit. I'd suggest changing `top: 0.3em;` to be: 

```
    top: 50%;
    transform: translateY(-50%);
```

Then we wouldn't need to adjust the padding at all.

**Comment by thibaudcolas** [comment]
This was caused by us having two sidebars in the codebase, which I have changed in #8310. Sorry @Marmik2003, that means this PR is no longer needed.

**Pull request closed without merging**